control-plane/agents/src/bin/core/controller/reconciler/node/nexus.rs

```diff
@@ -145,19 +145,56 @@ async fn check_and_drain_node(context: &PollContext, node_spec: &NodeSpec) -> Po
     }
     // Change the node state to "drained"
     if !move_failures {
-        if let Err(error) = context
-            .specs()
-            .set_node_drained(context.registry(), node_spec.id())
-            .await
-        {
-            tracing::error!(
-                %error,
-                node.id = node_id.as_str(),
-                "Failed to set node to state drained"
-            );
-            return PollResult::Err(error);
+        // get the list of volumes
```

Comment:
So I think here you probably want to keep a timestamp to start counting towards the timeout? Maybe we can have an in-memory timeout in the nexus draining state?